### ISSUE TITLE
Improvement: Image usage reporting 

### ISSUE THREAD

**Issue by khink** [wagtail/wagtail#3886]
Images might be rendered in several parts of the site (for example, a Product image that appears in listings of various layouts). 

It would be nice to have an overview of:

- All places where that image is being used. This would make it easier to check if the image can be deleted.
- All dimensions that the image is rendered in. This would would make it easier to detect cases where resizing gives undesirable results.

**Comment by khink** [comment]
Started during Bristol Sprint 2017.

Talked with @gasman, and the first part is rather easy: set `WAGTAIL_USAGE_COUNT_ENABLED = True` in your settings and you will get a link in the header of the Image edit view. (Documented in http://docs.wagtail.io/en/v1.13/advanced_topics/settings.html#usage-for-images-documents-and-snippets)

**Comment by khink** [comment]
As for the second part, showing all the ways an image is (or could be) rendered: I have toyed with a  new view for this (https://github.com/khink/wagtail/tree/feature/image-rendering-reporting) which shows all existing Renditions. The usability is quite limited and I will not create a PR for this. Reasons:

1. Renditions are not available immediately after Image upload, but only after the Image was rendered in a certain size. This could be changed if you create all required Renditions immediately after Image upload, but for that you'd have to keep track of all rendition parameters in use, as mentioned in https://github.com/wagtail/wagtail/issues/929
2. Renditions are not linked to a particular page or URL (as they might be created from any template), so we can't tell admins where they can see the image in that size.
3. "Older" renditions (i.e Renditions that are not shown anywhere in the site, b/c template changed aspect ratio, or focal point changed) would have to be filtered out.

To address the issue of getting a preview of all possible uses of the image in the site, i think the approach in https://github.com/wagtail/wagtail/issues/929 would be the way to go, but i don't think this needs to be in Wagtail core. It might be a nice add-on.

**Comment by khink** [comment]
#3954 shows a proof of concept how to make usage also list pages that have the image in a streamfield. Might be a nice add-on for Wagtail.

**Comment by khink** [comment]
In #3961, Bertrand started work to improve the performance of this, and made it more re-usable.
